torchani/data/new.py

```diff
@@ -0,0 +1,444 @@
+import numpy as np
+import torch
+import functools
+from ._pyanitools import anidataloader
+import pkbar
+
+
+class CacheDataset(torch.utils.data.Dataset):
```

Comment:
Could you add docstring for it, and update `docs/api.rst`?

Reply:
docs new pr

Reply:
Rename to `CachedDataset`

Reply:
No, docs should come in the same PR.